Modelica/Blocks/Math.mo

```diff
@@ -3505,13 +3505,8 @@ so one can plot the result directly as frequency lines.
       valid = true;
     else
       y2 = u1^2-u2^2;
-      if y2>=0 then
-        y=sqrt(y2);
-        valid = true;
-      else
-        y=0;
-        valid = false;
-      end if;
+      valid = (y2 >= 0);  
+      y = noEvent(if y2 >= 0 then sqrt(y2) else 0);
```

Comment:
What about?

```
valid = noEvent(y2 >= 0);
y = if valid then sqrt(y2) else 0;
```

Reply:
That won't work, I believe, because valid is a discrete variable, so its equation must occur in an event.

Reply:
A nice variation would be
  `valid = y2 >= 0;
   y = noEvent(if valid then sqrt(y2) else 0);`
Alas, that doesn't work (at least not on the Dymola version I tried).  I didn't expect it to, but cannot say exactly why it should fail.